src/main/java/seedu/address/logic/commands/StatisticsCommand.java

```diff
@@ -18,38 +18,78 @@ public class StatisticsCommand extends Command {
 
     public static final String COMMAND_WORD = "statistics";
 
-
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        // create a filtered list of all the predicate persons
-        // then create iteration
         ObservableList<Person> lastShownList = model.getFilteredPersonList();
         FilteredList<Person> fullContactList = new FilteredList<>(lastShownList);
-        int numOfApplicants = 0;
-        HashMap<JobCode, Integer> jobDict = new HashMap<JobCode, Integer>();
-        HashMap<Tag, Integer> tagDict = new HashMap<Tag, Integer>();
+
+        HashMap<JobCode, JobCodeStatistics> jobStatisticsMap = new HashMap<>();
+        calculateJobCodeStatistics(fullContactList, jobStatisticsMap);
+        String statisticsMessage = buildStatisticsMessage(jobStatisticsMap);
+        return new CommandResult(statisticsMessage);
+    }
+
+    private void calculateJobCodeStatistics(FilteredList<Person> fullContactList,
+                                            HashMap<JobCode, JobCodeStatistics> jobStatisticsMap) {
         for (Person person : fullContactList) {
-            numOfApplicants++;
-            JobCode jobCode = person.getJobCode();
-            jobDict.put(jobCode, jobDict.getOrDefault(jobCode, 0) + 1);
-            Tag tag = person.getTag();
-            tagDict.put(tag, tagDict.getOrDefault(tag, 0) + 1);
+            JobCode jobCode = person.getJobCode(); // get person job code
+            String tagCode = person.getTag().toString(); // get person's tag
+            JobCodeStatistics jobStats = jobStatisticsMap.getOrDefault(jobCode, new JobCodeStatistics());
+            jobStats.incrementTag(tagCode);
```

Comment:
Nice use of encapsulation by calling incrementTag. It would be helpful to have a comment within the JobCodeStatistics class explaining how this method handles duplicate tags and if there are any performance implications when called frequently.